docs/overview/deep_dive/formatting.rst

```diff
@@ -169,9 +169,22 @@ We have a GitHub action that runs:
 1. Every day at 08:00 UTC
 2. Manually invoked by making a comment with ``ivy-gardener`` on a PR
 
-The first action is to ensure that the code is always formatted correctly. The second action is to allow you to
-reformat your code directly from GitHub. This is useful if you didn't setup pre-commit correctly and you or one
-of our maintainers want to reformat your code without having to clone the repository.
+The first action is to ensure that the code in the whole codebase is always formatted correctly. The second action 
+is to reformat the files you changed in your PR directly on GitHub. This is useful in case if you didn't setup 
+pre-commit correctly or if you or one of our maintainers want to reformat your code remotely.
+
+Under the hood, when ``ivy-gardener`` is found in a comment, a ivy bot will trigger the same set of lint checks 
+as in the pre-commit process. Then the suggested changes produced in the checks will be applied automatically as
+a new commit if there is any. 
+
+However, it is possible for the linters run in the ivy-gardener and the github action every day to face formatting 
+errors that need human intervention like typos and uninitialized arguments. In this case, errors will be thrown 
+by the linters and by the lint checks that runs later, while the fixes for other normal errors will still be applied 
+as the ivy-gardener is passing properly.
+
+On the other hand, ivy-gardener itself can fail if the bot handling it (ivy-branch) can not apply the changes 
+suggested by the linters, for example, when it does not have access to edit the target branch. In this case, you 
+should check the logs of lint checks to manually resolve the formatting errors by commiting the changes yourself.
```

Comment:
Here, you can simply give the maintainers access (this is an option in GitHub UI)